src/lib/ui/thermostat/ThermostatCard.tsx

```diff
@@ -0,0 +1,167 @@
+    <div className='seam-thermostat-card'>
+      <Content deviceId={deviceId} />
+    </div>
+  )
+}
+
+function Content(props: { deviceId: string }): JSX.Element | null {
+  const { deviceId } = props
+
+  const { device } = useDevice({
+    device_id: deviceId,
+  })
+
+  const [temperatureUnit, setTemperatureUnit] = useState<
+    'fahrenheit' | 'celsius'
+  >('fahrenheit')
+
+  const toggleTemperatureScale = (): void => {
+    setTemperatureUnit(
+      temperatureUnit === 'fahrenheit' ? 'celsius' : 'fahrenheit'
+    )
+  }
+
+  if (device == null) {
+    return null
+  }
+
+  return (
+    <div className='seam-thermostat-card-content'>
+      <div className='seam-thermostat-card-image-wrap'>
+        <DeviceImage device={device} />
+      </div>
+      <div className='seam-thermostat-card-details'>
+        <div className='seam-thermostat-heading-wrap'>
+          <h4 className='seam-thermostat-card-heading'>
+            {device.properties.name}
+          </h4>
+          <button
+            onClick={toggleTemperatureScale}
+            className='seam-thermostat-temperature-toggle'
+          >
+            <span className='seam-thermostat-temperature-toggle-label'>
+              º{temperatureUnit.slice(0, 1).toUpperCase()}
+            </span>
+          </button>
+        </div>
+
+        <div className='seam-thermostat-properties'>
+          {'temperature_fahrenheit' in device.properties &&
+            'temperature_celsius' in device.properties && (
+              <>
+                <div className='seam-thermostat-property-block'>
+                  <p className='seam-thermostat-property-label'>
+                    {t.temperature}:
+                  </p>
+                </div>
+                <div className='seam-thermostat-property-block'>
+                  <p className='seam-thermostat-property-value'>
+                    {Math.trunc(
+                      Number(
+                        temperatureUnit === 'fahrenheit'
+                          ? device.properties.temperature_fahrenheit
+                          : device.properties.temperature_celsius
+                      )
+                    )}
+                    º
+                  </p>
+                  {'relative_humidity' in device.properties && (
+                    <>
+                      <p className='seam-thermostat-property-value'>|</p>
+                      <p className='seam-thermostat-property-label'>
+                        {t.humidity}:
+                      </p>
+                      <p className='seam-thermostat-property-value'>
+                        {Number(device.properties.relative_humidity) * 100}%
+                      </p>
+                    </>
+                  )}
+                </div>
+              </>
+            )}
+
+          {'current_climate_setting' in device.properties &&
+            'hvac_mode_setting' in
+              (device.properties
+                ?.current_climate_setting as ClimateSetting) && (
```

Comment:
I don't love the conditions here. It would be nice to have some additional generic typing on `useDevice` but idk if we wanted to do that atm

Reply:
Noted. This is part of a larger pattern I am hoping we can solve by eventually moving to `@seamapi/types`. For now, you can add some type guards to `src/lib/seam/devices/types.ts` with there conditions so we can organize the type situation and make sure to capture the different type variants upstream.

(I am considering any files names `types.ts` as an anti-pattern indicating some gap we are shimming for our upstream types).

Reply:
Extending on that, instead of or in addition to the type guards, you can improve the nesting situation by just making components that return null when passed devices it can't render. 

See for example https://github.com/seamapi/react/blob/main/src/lib/ui/device/LockStatus.tsx